File: app/src/main/java/com/nollpointer/dates/views/VisualizerView.java
package com.nollpointer.dates.views;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.view.View;

import androidx.annotation.Nullable;

public class VisualizerView extends View {
    public static final String TAG = "VisualizerView";

    public static final float AMPLITUDE_MAX_VALUE = 15f;
    public static final float AMPLITUDE_MIN_VALUE = 0f;

    private Paint linePaint;

    private float amplitude = AMPLITUDE_MIN_VALUE;

    private int width;
    private int height;

    private boolean isHidden = false;


    public VisualizerView(Context context, AttributeSet attrs) {
        super(context, attrs);
        linePaint = new Paint();
        linePaint.setColor(0xff448aff);
        linePaint.setStyle(Paint.Style.FILL);
    }

    @Override
    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
        super.onSizeChanged(w, h, oldw, oldh);
        width = w;
        height = h;
    }

    //выставление амплитуды
    public void setAmplitude(float amplitude) {
        this.amplitude = amplitude;
        invalidate();
    }

    public void hide() {
        this.amplitude = AMPLITUDE_MIN_VALUE;
        isHidden = true;
        invalidate();
    }

    public void show() {
        isHidden = false;
    }

    @Override
    protected void onDraw(Canvas canvas) {
        //super.onDraw(canvas);

        int halfWidth = width / 2;
        float delta = 0f;

        if (!isHidden)
            delta = halfWidth * (amplitude / AMPLITUDE_MAX_VALUE);


        for (int i = 0; i < height; i++)
            canvas.drawLine(halfWidth - delta, i, halfWidth + delta, i, linePaint);

    }
}

File: app/src/main/java/com/nollpointer/dates/ui/view/VisualizerView.kt
package com.nollpointer.dates.ui.view

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.util.AttributeSet
import android.view.View

/**
 * Кастомная View для отображения голосового вывода
 *
 * @author Onanov Aleksey (@onanov)
 */
class VisualizerView @JvmOverloads constructor(
        context: Context?,
        attrs: AttributeSet? = null,
        defStyleAttr: Int = 0) : View(context, attrs, defStyleAttr) {

    private val linePaint: Paint = Paint()
    private var amplitude = AMPLITUDE_MIN_VALUE
    private var currentWidth = 0
    private var currentHeight = 0
    private var isHidden = false

    init {
        linePaint.color = -0xbb7501
        linePaint.style = Paint.Style.FILL
    }

    override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
        super.onSizeChanged(w, h, oldw, oldh)
        currentWidth = w
        currentHeight = h
    }

    //выставление амплитуды
    fun setAmplitude(amplitude: Float) {
        this.amplitude = amplitude
        invalidate()
    }

    fun hide() {
        amplitude = AMPLITUDE_MIN_VALUE
        isHidden = true
        invalidate()
    }

    fun show() {
        isHidden = false
    }

    override fun onDraw(canvas: Canvas) { //super.onDraw(canvas);
        val halfWidth = currentWidth / 2
        var delta = 0f
        if (!isHidden) delta = halfWidth * (amplitude / AMPLITUDE_MAX_VALUE)
        for (i in 0 until height) canvas.drawLine(halfWidth - delta, i.toFloat(), halfWidth + delta, i.toFloat(), linePaint)
    }

    companion object {
        const val AMPLITUDE_MAX_VALUE = 15f
        const val AMPLITUDE_MIN_VALUE = 0f
    }

}